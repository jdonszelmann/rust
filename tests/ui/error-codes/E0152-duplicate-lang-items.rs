//! Validates the correct printing of E0152 in the case of duplicate "lang_item" function
//! definitions.
//!
//! Issue: <https://github.com/rust-lang/rust/issues/31788>

//@ normalize-stderr: "loaded from .*libcore-.*.rlib" -> "loaded from SYSROOT/libcore-*.rlib"
//@ dont-require-annotations: NOTE
#![feature(lang_items)]

extern crate core;

use core::panic::PanicInfo;

#[lang = "panic_cannot_unwind"]
fn panic_impl(info: &PanicInfo) -> ! {
    //~^ ERROR: found duplicate lang item `panic_cannot_unwind` [E0152]
    //~| NOTE first defined in crate `core`
    loop {}
}

fn main() {}
