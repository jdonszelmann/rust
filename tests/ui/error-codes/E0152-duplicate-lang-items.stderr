error[E0152]: found duplicate lang item `panic_cannot_unwind`
  --> $DIR/E0152-duplicate-lang-items.rs:15:1
   |
LL | / fn panic_impl(info: &PanicInfo) -> ! {
LL | |
LL | |
LL | |     loop {}
LL | | }
   | |_^
   |
   = note: the lang item is first defined in crate `core` (which `E0152_duplicate_lang_items` depends on)
   = note: first definition in `core` loaded from SYSROOT/libcore-*.rlib
   = note: second definition in the local crate (`E0152_duplicate_lang_items`)

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0152`.
